### ISSUE TITLE
[BUG] F-Droid build failed

### ISSUE THREAD

**Issue by linsui** [MetaCubeX/ClashMetaForAndroid#17]
### Describe the bug

```
# github.com/Dreamacro/clash/adapter/outbound
clash/adapter/outbound/vmess.go:277:69: undefined: packetaddr.NewBindClient
clash/adapter/outbound/vmess.go:287:69: undefined: packetaddr.NewBindClient
```

We are using https://packages.debian.org/bullseye-backports/golang-go go 1.19.

### To Reproduce

The complete log is https://gitlab.com/fdroid/fdroiddata/-/jobs/3267168931. We run `gradle assembleMeta-alphaRelease`.

### Device Info

- Device:
- ROM:
- Android Version:

### Application Info

- Version:
- APK filename:
- Distribution Channel:

### Configure File

```yml
None.
```

### Logs

```raw
https://gitlab.com/fdroid/fdroiddata/-/jobs/3267168931
```

**Comment by licaon-kter** [comment]
Fails with 1.18.3 too https://gitlab.com/fdroid/fdroiddata/-/jobs/3267113264#L2363

**Comment by cubemaze** [comment]
git submodule update --init --recursive
and check out android-open
gradle assembleMeta-alphaRelease

**Comment by linsui** [comment]
> check out android-open

What's this?

**Comment by licaon-kter** [comment]
> git submodule update --init --recursive

We already do this: https://gitlab.com/fdroid/fdroiddata/-/blob/67a0b937e543ca489ccfdc62a3218bd93c3a1b86/metadata/com.github.metacubex.clash.meta.yml#L42

**Comment by linsui** [comment]
Ah, yes, it's the submodule branch. It's checked out with `git submodule update --init --recursive`, right?

**Comment by cubemaze** [comment]
@linsui right

**Comment by licaon-kter** [comment]
But the submodules commit points to the `android-open` branch head already afaica

**Comment by cubemaze** [comment]
go mod tidy?

**Comment by linsui** [comment]
Run this before the build?

**Comment by cubemaze** [comment]
@linsui i think so

**Comment by linsui** [comment]
https://gitlab.com/linsui/fdroiddata/-/jobs/3330668458

I ran go mod tidy in `core/src/foss/golang` and it failed.

**Comment by cubemaze** [comment]
@linsui I don’t know what‘s going on.

**Comment by cubemaze** [comment]
@linsui Please refer to the workflow.

**Comment by linsui** [comment]
I cloned this repo with submodules. Then I ran `go mod tidy` in `core/src/foss/golang`.

**Comment by linsui** [comment]
The new version still failed to build.

<details>

```
# github.com/Dreamacro/clash/component/process
clash/component/process/process_linux.go:135:16: undefined: fmt.Appendf
note: module requires Go 1.19
# github.com/Dreamacro/clash/transport/tuic
clash/transport/tuic/client.go:53:21: undefined: atomic.Int64
clash/transport/tuic/client.go:54:21: undefined: atomic.Bool
clash/transport/tuic/conn.go:120:26: undefined: atomic.Bool
clash/transport/tuic/conn.go:210:22: invalid operation: q.writeClosed != nil (operator != not defined on untyped nil)
clash/transport/tuic/server.go:133:53: undefined: atomic.Bool
clash/transport/tuic/server.go:134:28: undefined: atomic.Bool
clash/transport/tuic/server.go:261:32: undefined: atomic.Bool
note: module requires Go 1.19
> Task :core:externalGolangBuildMetaAlphaReleaseArm64V8a FAILED
FAILURE: Build failed with an exception.
* What went wrong:
Execution failed for task ':core:externalGolangBuildMetaAlphaReleaseArm64V8a'.
> Process 'command 'go'' finished with non-zero exit value 2
* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
* Get more help at https://help.gradle.org/
BUILD FAILED in 5m 52s
95 actionable tasks: 95 executed
```

</details>

**Comment by cubemaze** [comment]
requires go 1.19

**Comment by linsui** [comment]
Just noticed that... Trying again.

**Comment by cubemaze** [comment]
Wait a few minutes and try again, currently fixing some issues

**Comment by linsui** [comment]
```
go: downloading github.com/sina-ghaderi/rabbitio v0.0.0-20220730151941-9ce26f4f872e
go: updates to go.mod needed; to update it:
	go mod tidy
> Task :core:externalGolangBuildMetaAlphaReleaseArm64V8a FAILED
FAILURE: Build failed with an exception.
* What went wrong:
Execution failed for task ':core:externalGolangBuildMetaAlphaReleaseArm64V8a'.
> Process 'command 'go'' finished with non-zero exit value 1
```

We got a similiar error again. Could you please take a look? Do we need to bump the go version? We are using 1.20.

**Comment by cubemaze** [comment]
@linsui Enter the foss project for go mod tidy and upload the mod.

**Comment by linsui** [comment]
The problem is fixed in 2.8.2... Thanks!

**Comment by licaon-kter** [comment]
2.8.3 too as https://github.com/MetaCubeX/ClashMetaForAndroid/tree/26a47020/core/src/foss/golang points to an invalid commit

ref: https://gitlab.com/fdroid/fdroiddata/-/commit/32277c7d73518c4d6aefee61be103b4a74910429
